Library/Homebrew/brew.rb

```diff
@@ -93,6 +93,15 @@ class MissingEnvironmentVariables < RuntimeError; end
     internal_cmd ||= begin
       internal_dev_cmd = Commands.valid_internal_dev_cmd?(cmd)
       if internal_dev_cmd && !Homebrew::EnvConfig.developer?
+        if ENV["HOMEBREW_DEV_CMD_RUN"].blank? && !Homebrew::EnvConfig.no_dev_cmd_message?
+          opoo <<~MESSAGE
+            #{Tty.bold}#{cmd}#{Tty.reset} is a developer command, so
+            Homebrew's developer mode has been automatically turned on.
+            To turn developer mode off, run #{Tty.bold}brew developer off#{Tty.reset}
```

Comment:
Just a thought here actually: is this going to be displayed every run or just once? If just once: can we set that state in the relevant tests? If every time: I think it should be a one-off thing like analytics/donation messages. Thoughts?

Reply:
Depends on what you mean by "every time". It will show every time that a dev cmd is run when `homebrew.devcmdrun` is not set in `.git/config`. So for most people, only once ever. However, if you `brew developer off` and then run a developer command again, it will be shown.

I did try to get rid of the message in the tests by running `git config --replace-all homebrew.devcmdrun true` before running the test in the hopes of not showing the message, but I couldn't get that to work

Reply:
Ah, you know what, I just realized why it didn't work... The check for whether `homebrew.devcmdrun` is set happens in `brew.sh` not `brew.rb`. The integration tests skip right to `brew.rb`, so when it checks for `HOMEBREW_DEV_CMD_RUN`, it's never set (because it's usually set in `brew.sh`.

So I could either re-read from the config file in `brew.rb` or just manually set `HOMEBREW_DEV_CMD_RUN` for integration tests.

Reply:
> just manually set `HOMEBREW_DEV_CMD_RUN` for integration tests.

I think this is worth doing.



> Depends on what you mean by "every time". It will show every time that a dev cmd is run when `homebrew.devcmdrun` is not set in `.git/config`. So for most people, only once ever. However, if you `brew developer off` and then run a developer command again, it will be shown.

Gotcha, perfect ❤️